### PR TITLE
Team_1 - Configures testing frame work and Coveralls

### PR DESCRIPTION
**Pull request by johanperjulius1**
-We are doing the rails setup.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,14 @@ gem 'sass-rails', '~> 5.0'
 gem 'turbolinks'
 gem 'uglifier', '>= 1.3.0'
 
+group :development, :test do
+  gem 'rspec-rails'
+  gem 'shoulda-matchers'
+  gem 'capybara'
+  gem 'coveralls', require: false
+  gem 'factory_bot_rails'
+end
```
Comment (emtalen) [inline Gemfile:24]: You need to add pry-rails as well

File: Gemfile [hunk 2]
```diff
@@ -16,6 +16,14 @@ gem 'sass-rails', '~> 5.0'
...
+
 group :development do
   gem 'pry-byebug'
```
Comment (emtalen) [inline Gemfile]: this should not be in the development, we use pry in test environment

File: config/application.rb [hunk 3]
```diff
@@ -1,40 +1,27 @@
...
     config.action_mailer.default_url_options = { host: 'ca-mailboxer.herokuapp.com' }
-
     config.generators do |generate|
       generate.helper false
       generate.assets false
       generate.view_specs false
+      generate.helper_specs false
+      generate.routing_specs false
+      generate.controller_specs false
       generate.test_framework :rspec
```
Comment (emtalen) [inline config/application.rb]: this line about test framework should not be here, we are turning of the generators here. 
And you should also turn off the request spec generator
`generate.request_specs false`

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  config.include FactoryBot::Syntax::Methods
+end
+Shoulda::Matchers.configure do |config|
+        config.integrate do |with|
+                with.test_framework :rspec
+                with.library :rails
+        end
+end
```
Comment (emtalen) [inline spec/rails_helper.rb:29]: You need to format the shoulda matchers configuration

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
+end
```
Comment (emtalen) [inline spec/spec_helper.rb:11]: Clean up the comments in this file please and require the rails_helper here or in your .rspec file as we discussed in the morning scrum

**Review by emtalen — CHANGES_REQUESTED** [review]
The title of the pr should explain what the pr is adding, so something like `Configures testing frame work and Coveralls` for example. 
And this time i will give you a pass but for the next PR you need to have a template in the description written in markdown that explains what the PR is about, what you have learned and what changes have been made and added

File: lib/tasks/ci.rake [hunk 6]
```diff
@@ -0,0 +1,9 @@
+unless Rails.env.production?
+  require 'rspec/core/rake_task'
+  require 'coveralls/rake/task'
+  Coveralls::RakeTask.new
+  namespace :ci do
+  desc 'Run all tests and generate a merged coverage report'
+  task tests: [:spec, 'coveralls:push'] 
+  end
+end
```
Comment (tochman) [inline lib/tasks/ci.rake]: Please format the code in this one

File: spec/features/user_can_login_spec.rb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+Rspec.describe 'user authentication', type: :feature do
+  let(:user) { FactoryBot.create(:user) }
+end
```
Comment (tochman) [inline spec/features/user_can_login_spec.rb]: This is out of scope. Please remove this file.

**Review by tochman — CHANGES_REQUESTED** [review]
Please address comments (and answer the comments - either with a response or an emoji). Also, from now you have to stick to the git-flow that we presented to you. 22 commits on a configuration PR is WAY too much.

File: spec/features/user_can_login_spec.rb (hunk 7 above)
Comment (johanperjulius1) [reply spec/features/user_can_login_spec.rb]: Hi! This should be removed now !

File: lib/tasks/ci.rake (hunk 6 above)
Comment (johanperjulius1) [reply lib/tasks/ci.rake]: This should be formatet!

**Review by tochman — APPROVED** [review]
